s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/S3SourceTask.java

```diff
@@ -133,11 +137,12 @@ private void initializeS3Client() {
 
     private void prepareReaderFromOffsetStorageReader() {
         sourceRecordIterator = new SourceRecordIterator(s3SourceConfig, s3Client, this.s3Bucket, offsetManager,
-                this.outputWriter, failedObjectKeys);
+                this.transformer, fileReader);
     }
 
     @Override
     public List<SourceRecord> poll() throws InterruptedException {
+        LOGGER.info("Polling again");
         synchronized (pollLock) {
             final List<SourceRecord> results = new ArrayList<>(s3SourceConfig.getInt(MAX_POLL_RECORDS));
 
```

Comment:
Flat sleep on error is not optimal, it should be some kind of exponential backoff.

Reply:
Logging DataException does not fail the task. This will result in dropped records.

Reply:
The exponential backoff system has an existing configuration, use it

Reply:
For DataException, we have another jira, to look at all corrupted records. That should address this, and any dropped records should be notified based on the configuration.

Reply:
For exponential backoff, I only found something similar in sink connector/context but not in source. Can you provide an example ?

Reply:
> For exponential backoff, I only found something similar in sink connector/context but not in source. Can you provide an example ?

It's just a concept, not a specific implementation.

`aws.s3.backoff.delay.ms` could be specific to the S3 client, so maybe it's not reusable.

Reply:
Further implementation wil be handled in ticket https://aiven.atlassian.net/browse/KCON-26